sdk/src/main/java/io/dapr/actors/client/ActorProxyImpl.java

```diff
@@ -125,8 +144,8 @@ public String getActorType() {
      * @param <T>      Expected response type.
      * @return Response object, null or RuntimeException.
      */
-    private <T> T unwrap(final String response, Class<T> clazz) {
-        return this.serializer.unwrapMethodResponse(response, clazz);
+    private <T> T unwrap(final String response, Class<T> clazz) throws IOException {
+        return this.serializer.deserialize(response, clazz);
```

Comment:
I wonder if this can actually work. There is a reason we use the unwrapMethodResponse - I used the example to manually validate end-to-end and that drove this change. I see that you are mocking serializer, so it might be why it does not fail in the unit tests. See my comment on using a real Serializer instance.

Reply:
The mock serializer is just for the constructor, the reason that I change this, is that doing some testing using the examples for .net-SDK, the response to call an actor method is not wrapped in a data node in the JSON response, that the unwrapMethodResponse expect inside the serializer object, please let me know if I am wrong in the assumption that the response does not contains  the "data" wrapper, or if I should change the method  unwrapMethodResponse in order to be used in both cases (with data wrapper and without data wrapper)

Reply:
Can you test this using the example for Actors? It will tell us if this will work or not.

Reply:
I already test this with the Actor example, you were right the response is a base64 under data, I already change the unit test, and the code to expect this kind of answers from Dapr. Thanks for your help.